htdocs/web_portal/controllers/downtime/add_downtime.php

```diff
@@ -158,62 +158,103 @@ function draw(\User $user = null) {
     }
 
-    // URL Mapping
-    // If the user wants to add a downtime to a specific site, show only that site's SEs
-    else if(isset($_GET['site'])) {
-        $site = \Factory::getSiteService()->getSite($_GET['site']);
-        if(\Factory::getRoleActionAuthorisationService()->authoriseAction(\Action::EDIT_OBJECT, $site, $user)->getGrantAction() == FALSE){
-           throw new \Exception("You don't have permission over $site");
-        }
-        $ses = $site->getServices();
-        $params = array('ses' => $ses, 'nowUtc' => $nowUtcDateTime->format('H:i T'), 'selectAll' => true);
-        show_view("downtime/add_downtime.php", $params);
-        die();
+    // URL Mapping for `site` and `se`
+    elseif (isset($_GET['site']) && isset($_GET['se'])) {
+        displaySiteAndSeEndpoints($user, $nowUtcDateTime);
     }
 
-    // URL Mapping
-    // If the user wants to add a downtime to a specific SE, show only that SE
-    else if(isset($_GET['se'])) {
-        $se = \Factory::getServiceService()->getService($_GET['se']);
-        $site = \Factory::getSiteService()->getSite($se->getParentSite()->getId());
-        if(\Factory::getRoleActionAuthorisationService()->authoriseAction(\Action::EDIT_OBJECT, $se->getParentSite(), $user)->getGrantAction() == FALSE){
-           throw new \Exception("You do not have permission over $se.");
-        }
+    // URL Mapping for `site` ONLY
+    elseif (isset($_GET['site'])) {
+       displaySiteEndpoints($user, $nowUtcDateTime);
+    }
 
-        //$ses = array($se);
-        $ses = $site->getServices();
-        $params = array('ses' => $ses, 'nowUtc' => $nowUtcDateTime->format('H:i T'), 'selectAll' => true);
-        show_view("downtime/add_downtime.php", $params);
-        die();
+    // URL Mapping for `se` ONLY
+    elseif (isset($_GET['se'])) {
+        displaySeEndpoints($user, $nowUtcDateTime);
     }
 
-    // If the user doesn't want to add a downtime to a specific SE or site show all SEs
+    // Generic URL mapping
     else {
-        $ses = array();
-        if($user->isAdmin()){
-            //If a user is an admin, return all SEs instead
-            $ses = \Factory::getServiceService()->getAllSesJoinParentSites();
-        } else {
-             //$allSites = \Factory::getUserService()->getSitesFromRoles($user);
-
-            // Get all ses where the user has a GRANTED role over one of its
-            // parent OwnedObjects (includes Site and NGI but not currently Project)
-            $sesAll = \Factory::getRoleService()->getReachableServicesFromOwnedObjectRoles($user);
-            // drop the ses where the user does not have edit permissions over
-            foreach($sesAll as $se){
-                if(\Factory::getRoleActionAuthorisationService()->authoriseAction(\Action::EDIT_OBJECT, $se->getParentSite(), $user)->getGrantAction() ){
-                    $ses[] = $se;
-                }
+      displayAllServiceEndpoints($user, $nowUtcDateTime);
```

Comment:
```suggestion
        displayAllServiceEndpoints($user, $nowUtcDateTime);
```
indent only 6 spaces, not 8